beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/HistoricalBatchFetcher.java

```diff
@@ -223,7 +223,8 @@ private SafeFuture<Boolean> requestBlocksAndBlobSidecarsByRange() {
             requestParams.getStartSlot(), requestParams.getCount(), requestManager::processBlock);
 
     final SafeFuture<Void> blobSidecarsRequest;
-    if (blobSidecarManager.isAvailabilityRequiredAtSlot(endSlot)) {
+    if (blobSidecarManager.isAvailabilityRequiredAtSlot(endSlot)
+        || blobSidecarManager.isAvailabilityRequiredAtSlot(requestParams.getStartSlot())) {
```

Comment:
Couldn't the `[startSlot, endSlot]` period cross the boundary of Deneb -> Eip7594  transition?

Reply:
Oh ok, it looks like this case should be handled correctly: for both Deneb boundaries remote peer should just return the blobs for Deneb period and ignore others which are outside of Deneb period